[reginateh] iP

reginateh opened the pull request:
# Otto :owl:
> Otto would rather be napping, but he supposes he can help you with your tasks.

Otto helps you remember things you need to do, although he doesn't want to. He is, 

- text-based
- easy to learn
- ~~FAST~~ *SUPER FAST* to use

All you need to do is,

1. download it from [here](https://nus-cs2103-ay2425s1.github.io/website/schedule/week4/project.html).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😄 

## Features

 - [x] Managing tasks
 - [ ] Managing deadlines (coming soon)
 - [ ] Reminders (coming soon)

---
If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:
```Java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

reginateh committed:
Change Duke to Otto

reginateh committed:
Add Otto

reginateh committed:
Otto.java: Add user input and echo

reginateh committed:
Otto.java: Change static methods to instance methods

reginateh committed:
Add new classes Task and TaskList

reginateh committed:
Otto.java: Create list and add task

reginateh committed:
Add completion status in Task & update TaskList index

reginateh committed:
Add mark and unmark commands

reginateh committed:
Add 3 subclasses of Task: Todo, Deadline, Event

reginateh committed:
Add 3 actions to create 3 types of tasks

reginateh committed:
Modify test script and add testcases

reginateh committed:
Add OttoException and handle input errors

reginateh committed:
Format code

reginateh committed:
Add the delete command

reginateh committed:
Add OttoResponse.java to store response constants

mattisongjj commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,19 @@
+abstract public class Task {
```
I like the use of an abstract Task class that all other tasks inherit from, makes it easier in the case we want to add future tasks

mattisongjj replied:
Perhaps this can be further abstracted by placing all relevant code in a 'task' package

mattisongjj commented on src/main/java/Otto.java:
```diff
@@ -0,0 +1,113 @@
+        printLine();
+    }
+
+    private void displayTaskList() {
```
Perhaps tasks related methods could be placed in the TaskList.java file rather than in this file

mattisongjj commented on src/main/java/OttoResponses.java:
```diff
@@ -0,0 +1,27 @@
+public final class OttoResponses {
```
I like the use of a class that encapsulates the responses

mattisongjj replied:
Could look into changing the names of the variables here to screaming snake case, since the otto responses are constant

mattisongjj reviewed (commented):
LGTM

L-rrrr commented on src/main/java/Otto.java:
```diff
@@ -0,0 +1,113 @@
+            return;
+        }
+        String[] command = userInput.split("\\s+", 2);
+        try {
+            switch (command[0].toLowerCase()) {
+                case "list":
+                    this.displayTaskList();
+                    break;
+                case "mark":
+                    this.markComplete(Parser.parseMarkComplete(command), true);
+                    break;
+                case "unmark":
+                    this.markComplete(Parser.parseMarkComplete(command), false);
+                    break;
+                case "todo":
+                    this.addTask(Parser.parseTask("todo", userInput));
+                    break;
+                case "deadline":
+                    this.addTask(Parser.parseTask("deadline", userInput));
+                    break;
+                case "event":
+                    this.addTask(Parser.parseTask("event", userInput));
+                    break;
+                case "delete":
+                    this.deleteTask(Parser.parseDeleteTask(command));
+                    break;
+                default:
+                    throw new OttoException(OttoResponses.unknown);
+            }
+        } catch (OttoException e) {
+            this.displayMsg(e.getMessage());
+        }
+    }
```
I like how you follow the coding standard and the Egyptian style to make the code more readable

L-rrrr commented on src/main/java/OttoResponses.java:
```diff
@@ -0,0 +1,27 @@
+    public static final String markError = "You need to tell Otto what task you want to mark. Not like Otto cares though.";
+    public static final String eventError = "The format for event is wrong. Missing description, start time or end time.";
+    public static final String deadlineError = "The format for deadline is wrong. Missing description or deadline.";
+    public static final String todoError = "Todo must contain a description.";
```
I like that you make the strings final here so that they cannot be modified in other classes

L-rrrr commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,19 @@
+abstract public class Task {
+    private String description;
+    private boolean isComplete;
```
I like how you name your boolean field

L-rrrr commented on src/main/java/Parser.java:
```diff
@@ -0,0 +1,71 @@
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
```
I like how you import and there is no use of wildcard

L-rrrr reviewed (commented):
Looks all good!
Perhaps can add some javadoc for different class and methods.

reginateh committed:
Add storage methods to store tasks in txt file

reginateh committed:
Add LocalDate for Deadline tasks

reginateh committed:
Add Dateformatter to Print formatted date

reginateh committed:
Merge branch 'branch-Level-8'

reginateh committed:
Add Ui class to handle terminal outputs

reginateh committed:
Add more Otto responses

reginateh committed:
Make Otto a singleton

reginateh committed:
Update display of file i/o errors

reginateh committed:
Merge branch 'branch-A-MoreOOP'

reginateh committed:
Move all files into package otto

reginateh committed:
Merge branch 'branch-A-Packages'

reginateh committed:
Merge branch 'add-gradle-support'

reginateh committed:
Add Gradle

reginateh committed:
Add ParserTest

reginateh committed:
Add TaskListTest

reginateh committed:
Update build.gradle

reginateh committed:
Merge branch 'branch-A-JUnit'

reginateh committed:
Add JavaDoc

reginateh committed:
Format code

reginateh committed:
Resolve merge conflicts

reginateh committed:
Add find command

reginateh committed:
Add JavaFX GUI

reginateh committed:
Update tests

reginateh committed:
Merge branch 'branch-Level-10'

reginateh committed:
Add assertions to the codebase

Add assertions to files: Otto.java, Storage.java, TaskList.java
to check for null values, invalid inputs, and other assumptions.

Modify build.gradle to run the assertions.

reginateh committed:
Merge pull request #1 from reginateh/branch-A-Assertions

Add assertions to the codebase

reginateh committed:
Improve code quality
- Extract date parsing logic to DateUtils.java
- Extract constants in methods as final fields
- Remove unused imports

reginateh committed:
Rename constants in OttoResponses to SCREAMING_SNAKE_CASE

reginateh committed:
TaskList.java: Edit JavaDoc format

reginateh committed:
Merge pull request #3 from reginateh/branch-A-CodeQuality

Improve code quality

reginateh committed:
Add new files Tag.java and TagList.java

reginateh committed:
Add tagging feature
- Add new field "tags" to Task class
- Add new classes Tag and TagList
- Update parser to parse commands with tags
- Add new command to find tasks with a specific Tag
- Update tests

reginateh committed:
Merge pull request #4 from reginateh/branch-C-tagging

Add tagging feature

reginateh committed:
Add Ui.png and update README.md

reginateh committed:
Set window title to Otto

reginateh committed:
Update Ui.png

reginateh committed:
Fix date parsing bug

reginateh committed:
Update README.md

reginateh committed:
Update README.md
